backend/private-graph/graph/resolver.go

```diff
@@ -636,9 +635,44 @@ func (r *Resolver) SetErrorFrequenciesInflux(ctx context.Context, projectID int,
 			Value        int64
 		}{}
 		errorGroupMap[strconv.Itoa(errorGroup.ID)] = errorGroup
-		errorGroupIDs = append(errorGroupIDs, errorGroup.ID)
 	}
-	results, err := r.GetErrorGroupFrequencies(ctx, projectID, errorGroupIDs, params, "")
+	var oldErrorGroupIDs []int
+	var err error
+	var results []*modelInputs.ErrorDistributionItem
+	for _, eg := range errorGroups {
+		if time.Since(eg.CreatedAt) <= time.Hour {
+			results, err = r.GetErrorGroupFrequenciesUnsampled(ctx, eg.ProjectID, eg.ID)
+			if err != nil {
+				log.WithContext(ctx).Error(err)
+			}
+			endDate := time.Now().UTC().AddDate(0, 0, -1)
+			startDate := endDate.AddDate(0, 0, -1*(lookbackPeriod-1))
+			for curDate := startDate; !curDate.After(endDate); curDate = curDate.AddDate(0, 0, 1) {
+				eg.ErrorFrequency = append(eg.ErrorFrequency, 0)
+				eg.ErrorMetrics = append(eg.ErrorMetrics, &struct {
+					ErrorGroupID int
+					Date         time.Time
+					Name         string
+					Value        int64
+				}{ErrorGroupID: eg.ID, Date: curDate.Truncate(24 * time.Hour), Name: "count", Value: 0})
+			}
+			for _, r := range results {
+				if r.Name == "count" {
+					eg.ErrorFrequency = append(eg.ErrorFrequency, r.Value)
+				}
+				eg.ErrorMetrics = append(eg.ErrorMetrics, &struct {
+					ErrorGroupID int
+					Date         time.Time
+					Name         string
+					Value        int64
+				}{ErrorGroupID: eg.ID, Date: r.Date, Name: r.Name, Value: r.Value})
+			}
```

Comment:
It looks like the above block of code is setting unsampled data if the error group was created in the last hour.
Is there a reason why we let the code continue from here? Why not return early? 
I _think_ it might not matter because `oldErrorGroupIDs` will be empty but wouldn't we be able to scope that variable better if we returned early?

Reply:
since `errorGroups` may consist of more than one error group of varying age, we still want to build up the `oldErrorGroupIDs` for ones that are older than an hour that can leverage the other aggregation method. the final result could be a set of frequencies for both old and new error groups.